packages/tests/server/react/polymorphism.test.tsx

```diff
@@ -202,21 +259,33 @@ describe('polymorphism', () => {
     });
 
     test('can use the abstract interface with a factory instance which has been merged with some extra procedures', async () => {
-      const { trpc } = ctx;
+      const { useTRPC } = ctx;
 
       function DiscussionsExportPage() {
-        const utils = trpc.useUtils();
+        const trpc = useTRPC();
 
         const [currentExport, setCurrentExport] = useState<number | null>(null);
+
+        // TODO: clean this back up
         const invalidate = useMutation({
-          mutationFn: () => utils.invalidate(),
+          // Option A: fully explicit
+          // mutationFn: () =>
+          //   trpc.queryClient.invalidateQueries({
+          //     queryKey: trpc.github.getQueryKey(),
+          //   }),
+
+          // Option B: QueryFilter factory, re-usable in multiple places
+          // mutationFn: () =>
+          //   trpc.queryClient.invalidateQueries(trpc.github.getQueryFilter()),
+
+          // // Option C: invalidate method, simple, but cancellation, refetches etc would need their own
+          mutationFn: () => trpc.github.invalidate(),
```

Comment:
A few examples here of possible APIs

```tsx
const invalidate = useMutation({
  // Option A: fully explicit
  mutationFn: () =>
    trpc.queryClient.invalidateQueries({
      queryKey: trpc.github.getQueryKey(),
    }),

  // Option B: QueryFilter factory, re-usable in multiple places
  mutationFn: () =>
    trpc.queryClient.invalidateQueries(trpc.github.getQueryFilter()),

  // // Option C: invalidate method, simple, but cancellation, refetches etc would need their own
  mutationFn: () => trpc.github.invalidate(),
});
```

Option B is probably the right one, it's actually quite re-usable which is interesting, as invalidate isn't trhe only thing to take QueryFilters, so keeps our API surface down